compositor_render/src/scene/layout.rs

```diff
@@ -161,18 +161,28 @@ impl StatefulLayoutComponent {
         child: &mut StatefulComponent,
         position: AbsolutePosition,
         parent_size: Size,
+        child_padding: Padding,
+        parent_padding: Padding,
         pts: Duration,
     ) -> NestedLayout {
-        let width = position.width.unwrap_or(parent_size.width);
-        let height = position.height.unwrap_or(parent_size.height);
+        let width = position
+            .width
+            .unwrap_or(parent_size.width + child_padding.horizontal());
+        let height = position
+            .height
+            .unwrap_or(parent_size.height + child_padding.vertical());
 
         let top = match position.position_vertical {
-            VerticalPosition::TopOffset(top) => top,
-            VerticalPosition::BottomOffset(bottom) => parent_size.height - bottom - height,
+            VerticalPosition::TopOffset(top) => top + parent_padding.top,
+            VerticalPosition::BottomOffset(bottom) => {
+                parent_size.height - bottom - height - parent_padding.bottom + parent_padding.top
+            }
         };
         let left = match position.position_horizontal {
-            HorizontalPosition::LeftOffset(left) => left,
-            HorizontalPosition::RightOffset(right) => parent_size.width - right - width,
+            HorizontalPosition::LeftOffset(left) => left + parent_padding.left,
+            HorizontalPosition::RightOffset(right) => {
+                parent_size.width - right - width - parent_padding.right + parent_padding.left
```

Comment:
why parent padding is taken into account? parent padding should not have any affect on absolutely positioned element

Reply:
The parent padding does affect the absolutely positioned child element. At least that's the case in CSS's `position: relative` and IIRC ours `absolute` is modeled after `position: relative` unless I misunderstood it